src/Explorer/Controls/Settings/SettingsComponent.tsx

```diff
@@ -0,0 +1,1285 @@
+import * as React from "react";
+import * as AutoPilotUtils from "../../../Utils/AutoPilotUtils";
+import * as Constants from "../../../Common/Constants";
+import * as DataModels from "../../../Contracts/DataModels";
+import * as SharedConstants from "../../../Shared/Constants";
+import * as ViewModels from "../../../Contracts/ViewModels";
+import DiscardIcon from "../../../../images/discard.svg";
+import SaveIcon from "../../../../images/save-cosmos.svg";
+import InfoColor from "../../../../images/info_color.svg";
+import Warning from "../../../../images/warning.svg";
+import { traceStart, traceFailure, traceSuccess } from "../../../Shared/Telemetry/TelemetryProcessor";
+import { Action } from "../../../Shared/Telemetry/TelemetryConstants";
+import { RequestOptions } from "@azure/cosmos/dist-esm";
+import Explorer from "../../Explorer";
+import { updateOffer } from "../../../Common/DocumentClientUtilityBase";
+import { updateCollection } from "../../../Common/dataAccess/updateCollection";
+import { CommandButtonComponentProps } from "../../Controls/CommandButton/CommandButtonComponent";
+import { userContext } from "../../../UserContext";
+import { updateOfferThroughputBeyondLimit } from "../../../Common/dataAccess/updateOfferThroughputBeyondLimit";
+import SettingsTab from "../../Tabs/SettingsTabV2";
+import {
+  manualToAutoscaleDisclaimerElement,
+  ttlWarning,
+  indexingPolicyTTLWarningMessage,
+  updateThroughputBeyondLimitWarningMessage,
+  updateThroughputDelayedApplyWarningMessage,
+  getThroughputApplyDelayedMessage,
+  getThroughputApplyShortDelayMessage,
+  getThroughputApplyLongDelayMessage
+} from "./SettingsRenderUtils";
+import { ScaleComponent } from "./SettingsSubComponents/ScaleComponent";
+import {
+  getMaxRUs,
+  getMinRUs,
+  hasDatabaseSharedThroughput,
+  canThroughputExceedMaximumValue,
+  GeospatialConfigType,
+  TtlType,
+  ChangeFeedPolicyState,
+  SettingsV2TabTypes,
+  getTabTitle,
+  isDirty,
+  TtlOff,
+  TtlOn,
+  TtlOnNoDefault
+} from "./SettingsUtils";
+import { ConflictResolutionComponent } from "./SettingsSubComponents/ConflictResolutionComponent";
+import { SubSettingsComponent } from "./SettingsSubComponents/SubSettingsComponent";
+import { Pivot, PivotItem, IPivotProps, IPivotItemProps, IChoiceGroupOption, Image } from "office-ui-fabric-react";
+import "./SettingsComponent.less";
+import { IndexingPolicyComponent } from "./SettingsSubComponents/IndexingPolicyComponent";
+
+interface SettingsV2TabInfo {
+  tab: SettingsV2TabTypes;
+  content: JSX.Element;
+}
+
+interface ButtonV2 {
+  isVisible: () => boolean;
+  isEnabled: () => boolean;
+  isSelected?: () => boolean;
+}
+
+export interface SettingsComponentProps {
+  settingsTab: SettingsTab;
+}
+
+export interface SettingsComponentState {
```

Comment:
>SettingsComponentState [](start = 17, length = 22)

As discussed, let us have 4 values; scaleUpdated, indexingPolicyUpdated, ConflictresolnUpdated etc which this component will use to determine in enabling or disabling save /discard buttons.  And the associated logic for scaleUpdated is all within the scale component.